src/js/binaryen.js-post.js

```diff
@@ -1162,6 +1162,14 @@ Module['Module'] = function(module) {
     out = old;
     return ret;
   };
+  this['emitStackIR'] = function() {
+    var old = out;
+    var ret = '';
+    out = function(x) { ret += x + '\n' };
+    this['runPasses'](['print-stack-ir']);
```

Comment:
you also need `generate-stack-ir` before it. it may work without it, if stack IR already exists, which is the case if you ran the right optimizations.

Reply:
I see. What about "optimize-stack-ir", should this pass be run as well?

Reply:
Ah, good question - I was going to say no, that optimization should be done separately, but then there's a problem, if you optimize before, then calling generate here would trample that.

So maybe two options to pick from here:

1. `emitStackIR`, `emitOptimizedStackIR`, two separate functions. Stack IR is kind of special so maybe this is justified.
2. Have `emitStackIR` just run print, and not generate. Then the user must previously do either `generate` or `generate,optimize` manually. That seems more error-prone I think, so I lean towards option 1.

Reply:
Currently, the PR does just 2., but I also like 1. for simplicitly, going to add this :)

Reply:
Decided to go with an additional `optimize` parameter on the function because it's otherwise the same, hope that's ok :) Also added a test.